tests/README.md

```diff
@@ -1,25 +1,239 @@
-Mender Integration testing
-===================
+# Mender Integration Testing
 
+<!-- markdown-toc start - Don't edit this section. Run M-x markdown-toc-refresh-toc -->
+**Table of Contents**
 
-####Requirements
+- [Mender Integration Testing](#mender-integration-testing)
+    - [Getting Started](#getting-started)
+    - [Installing Dependencies](#installing-dependencies)
+        - [Debian](#debian)
+        - [Alpine Linux](#alpine-linux)
+        - [Python2 and Python3](#python2-and-python3)
+    - [Running the Tests Locally](#running-the-tests-locally)
+        - [Isolating Python Dependencies](#isolating-python-dependencies)
+            - [Install](#install)
+            - [Initialize the Virtual Environment](#initialize-the-virtual-environment)
+                - [(Optional) -- Select Python Version](#optional----select-python-version)
+            - [Activate the Virtual Environment](#activate-the-virtual-environment)
+            - [Install the Python Dependencies from the Requirements File](#install-the-python-dependencies-from-the-requirements-file)
+        - [Running the Tests](#running-the-tests)
+    - [Modifying the Docker Images Employed](#modifying-the-docker-images-employed)
+        - [Example -- Running with a Custom Backend Service](#example----running-with-a-custom-backend-service)
+        - [Example -- Running with a Custom Client](#example----running-with-a-custom-client)
+    - [Running the Tests in a VM](#running-the-tests-in-a-vm)
+        - [Vagrant](#vagrant)
+- [Legacy Stuff (TODO -- Remove)](#legacy-stuff-todo----remove)
+    - [-](#--1)
+    - [OS X](#os-x)
+    - [Known issues](#known-issues)
+    - [Tips and Tricks](#tips-and-tricks)
+
+<!-- markdown-toc end -->
+
+
+-------------------------------------------------------------------------------
+
+
+## Getting Started
+
+The dependencies for the integration tests are collected and organized in
+dependency files in the `./requirements` folder, and separated into
```

Comment:
Actually they are in `requirements/integration`, and I don't think we need the second level.

Reply:
Damn, thought I had gotten them all :p

Reply:
On the second level one, I initially started out with just one level, until I read the `.gitlab-ci.yml` file. The dependencies there are -- as far as I could understand -- sort of ordered this way, where say, the acceptance tests have different dependencies, and hence I thought it natural to split it this way, and change most of them here. Then again, maybe we should just keep this for the integration tests?

Reply:
Well, mind the different repos here.

It is true that `.gitlab-ci.yml` in `mender-qa` differentiate that way, but actually the requirements are coming from different repos, right? So only integration tests reqs come from `integration` repo (the one we are now reviewing in this PR), while the acceptance tests requirements come from `meta-mender` repo. Hence to make here this second level I don't see it useful.

Reply:
Or from a simpler point of view: this `tests` folder already implies "integration tests". Other test suites (backend-tests or acceptance tests) live in different folders and repos, and only there shall they have their requirements listed (if necessary).

Reply:
Yes, you are right! Thx. Removing.